piptools/scripts/compile.py

```diff
@@ -78,15 +79,15 @@ def cli(verbose, quiet, dry_run, pre, rebuild, find_links, index_url, extra_inde
             src_files = (DEFAULT_REQUIREMENTS_FILE,)
         elif os.path.exists('setup.py'):
             src_files = ('setup.py',)
-            if not output_file:
-                output_file = 'requirements.txt'
         else:
             raise click.BadParameter(("If you do not specify an input file, "
                                       "the default is {} or setup.py").format(DEFAULT_REQUIREMENTS_FILE))
 
-    if len(src_files) == 1 and src_files[0] == '-':
-        if not output_file:
+    if len(src_files) == 1:
+        if src_files[0] == '-' and not output_file:
             raise click.BadParameter('--output-file is required if input is from stdin')
+        elif src_files[0] == 'setup.py':
```

Comment:
`src_files == ('setup.py',)` ?

Reply:
I like that. No need for `len(src_files) == 1` then.

Reply:
@blueyed i've fixed that block, see the latest commit please. Is it okay?